TestJWT Failure in Vault 1.11

Vault 1.11 introduced the following errors in test_jwt.py. Creating an issue to track the resolution of this failure.

```
FAILED tests/integration_tests/api/auth_methods/test_jwt.py::TestJWT::test_create_role_0_success - AssertionError: 'data' not found in <Response [204]>
```

--- a/tests/integration_tests/api/auth_methods/test_jwt.py
+++ b/tests/integration_tests/api/auth_methods/test_jwt.py
@@ -108,10 +108,16 @@ class TestJWT(HvacIntegrationTestCase, TestCase):
             path=self.TEST_JWT_PATH,
         )
         logging.debug("create_role response: %s" % response)
-        self.assertIn(
-            member="data",
-            container=response,
-        )
+        if utils.vault_version_lt("1.11"):
+            self.assertIn(
+                member="data",
+                container=response,
+            )
+        else:
+            self.assertEqual(
+                204,
+                response.status_code,
+            )
 
     @parameterized.expand(
         [

